src/main/java/duke/DateTimeParser.java

```diff
@@ -0,0 +1,127 @@
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+
+/**
+ * Contains functionality for parsing date and time input, as well as formatting
+ * LocalDate and LocalDateTime objects.
+ */
+public class DateTimeParser {
+
+    private static DateTimeFormatter parser = new DateTimeFormatterBuilder()
+            .parseCaseInsensitive()
+            .appendOptional(DateTimeFormatter.ofPattern("d-M-yyyy H:mm"))
+            .appendOptional(DateTimeFormatter.ofPattern("d-M-yyyy h:mma"))
+            .appendOptional(DateTimeFormatter.ofPattern("d-M-yyyy"))
+            .appendOptional(DateTimeFormatter.ofPattern("d/M/yyyy H:mm"))
+            .appendOptional(DateTimeFormatter.ofPattern("d/M/yyyy h:mma"))
+            .appendOptional(DateTimeFormatter.ofPattern("d/M/yyyy"))
+            .appendOptional(DateTimeFormatter.ofPattern("d MMM yyyy H:mm"))
+            .appendOptional(DateTimeFormatter.ofPattern("d MMM yyyy h:mma"))
+            .appendOptional(DateTimeFormatter.ofPattern("d MMM yyyy"))
+            .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+            .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0).toFormatter();
+
+    private static DateTimeFormatter dateStorageFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+
+    private static DateTimeFormatter dateTimeStorageFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+
+    private static DateTimeFormatter dateDisplayFormatter = DateTimeFormatter.ofPattern("MMMM yyyy");
+
+
+    /**
+     * Returns a LocalDateTime object from the given string containing date and time.
+     * The format for the date and time string is "dd-mm-yyyy hh:ss". The time may be omitted.
+     *
+     * @param dateTimeString a date and time string with the format "dd-mm-yyyy hh:ss"
```

Comment:
Very minor nit here but according to the [coding standard for Javadocs](https://se-education.org/guides/conventions/java/intermediate.html#comments), each parameter description should be punctuated. Perhaps you could add in a period at the end and capitalize the first letter after the parameter (`dateTimeString A date and time string with the format "dd-mm-yyyy hh:ss".`)